src/Middleware/IpFilter.php

```diff
@@ -29,7 +37,11 @@ public function __construct(string $allowedIp, ResponseFactoryInterface $respons
      */
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        if ($request->getServerParams()['REMOTE_ADDR'] !== $this->allowedIp) {
+        $clientIp = $request->getServerParams()['REMOTE_ADDR'] ?? null;
```

Comment:
Is it possible to make it play nice with https://github.com/yiisoft/yii-web/blob/master/src/Middleware/TrustedHostsNetworkResolver.php?

Reply:
Works with Network Resolvers:
```php
addMiddlewares([
  // ...
  new TrustedNetworkResolver(...)...,
  new IpFilter(new Ip()..., $factory, 'requestClientIp'),
  // ...
]);
// or without trusted network resolver
addMiddlewares([
  // ...
  new IpFilter(new Ip()..., $factory),    // In this case, the REMOTE_ADDR is used.
  // ...
])
// or in another PSR compatible framework, or custom IP attribute
addMiddlewares([
  // ...
  new IpFilter(new Ip()..., $factory, 'customIpAttribute'),
  // ...
]);
```

https://github.com/yiisoft/yii-web/pull/180/files#diff-9b136b6d3ef91d39bce71e8c20b80101R42

Reply:
Oh :) Thanks.